### PR TITLE
Edit delete command

### PR DESCRIPTION
**Pull request by SherniceSng**
`delete INDEX` will directly delete the contact of that corresponding index

`delete KEYWORD`:

- If `KEYWORD `is a full name or partial name that is unique to that contact, the contact will be deleted directly.
- If `KEYWORD `is not unique to any contact, a list of contacts that is filtered using `KEYWORD `will be given to the user, allowing them to see the index of each contact in the filtered list. From there, the user needs to give a `delete INDEX` command to specify which of the contacts they want to delete.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java [hunk 1]
```diff
@@ -62,7 +122,23 @@ public boolean equals(Object other) {
...
+                && predicate == null && otherDeleteCommand.predicate == null) {
+            return targetIndex.equals(otherDeleteCommand.targetIndex);
+        }
+
+        if (targetIndex == null && otherDeleteCommand.targetIndex == null
+                && predicate != null && otherDeleteCommand.predicate != null) {
+            return predicate.equals(otherDeleteCommand.predicate);
+        }
+
+        return false;
```
Comment (Droas590) [inline src/main/java/seedu/address/logic/commands/DeleteCommand.java:141]: the if statement here is a bit confusing, would be better if can add comments on this or make the code clearer like this
```
 // Both commands have null fields
    boolean bothHaveNullTargets = targetIndex == null && otherDeleteCommand.targetIndex == null;
    boolean bothHaveNullPredicates = predicate == null && otherDeleteCommand.predicate == null;
    
    // Both commands have non-null fields
    boolean bothHaveTargets = targetIndex != null && otherDeleteCommand.targetIndex != null;
    boolean bothHavePredicates = predicate != null && otherDeleteCommand.predicate != null;
    
    // Case 1: Both have null targetIndex and null predicate
    if (bothHaveNullTargets && bothHaveNullPredicates) {
        return true;
    }
    
    // Case 2: Both have targetIndex but null predicate
    if (bothHaveTargets && bothHaveNullPredicates) {
        return targetIndex.equals(otherDeleteCommand.targetIndex);
    }
    
    // Case 3: Both have null targetIndex but have predicate
    if (bothHaveNullTargets && bothHavePredicates) {
        return predicate.equals(otherDeleteCommand.predicate);
    }
    
    // All other cases are false
    return false;
```

**Review by Droas590 — COMMENTED** [review]
the equal method can be clearer, otherwise LGTM

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java (hunk 1 above)
Comment (SherniceSng) [reply src/main/java/seedu/address/logic/commands/DeleteCommand.java:141]: Okay, will make the changes. Thank you!

File: src/test/java/seedu/address/logic/parser/AddressBookParserTest.java [hunk 2]
```diff
@@ -50,9 +50,16 @@ public void parseCommand_clear() throws Exception {
 
     @Test
     public void parseCommand_delete() throws Exception {
```
Comment (VedJoshi) [inline src/test/java/seedu/address/logic/parser/AddressBookParserTest.java:52]: can have 2 separate tests? parseCommand_deleteByIndex() and parseCommand_deleteByKeyword()?, easier to test and maintain also.

**Review by VedJoshi — APPROVED** [review]

**Pull request merged**